docs/architecture/adr-042-group-module.md

```diff
@@ -0,0 +1,375 @@
+
+We propose merging the `x/group` module with its supporting [ORM/Table Store package](https://github.com/regen-network/regen-ledger/tree/master/orm) ([#7098](https://github.com/cosmos/cosmos-sdk/issues/7098)) into the Cosmos SDK and continuing development here. There will be a dedicated ADR for the ORM package.
+
+### Group
+
+A group is an aggregation of accounts with associated weights. It is not
+an account and doesn't have a balance. It doesn't in and of itself have any
+sort of voting or decision weight. It has an `admin` account which can manage members in the group.
+
+Groups are stored in state as part of an ORM-based `groupTable` using the `GroupInfo` type. The `group_id` is an auto-increment integer.
+
+```proto
+message GroupInfo {
+
+    // group_id is the unique ID of the group.
+    uint64 group_id = 1;
+
+    // admin is the account address of the group's admin.
+    string admin = 2;
+    
+    // metadata is any arbitrary metadata to attached to the group.
+    bytes metadata = 3;
+
+    // version is used to track changes to a group's membership structure that
+    // would break existing proposals. Whenever any members weight is changed,
+    // or any member is added or removed this version is incremented and will
+    // cause proposals based on older versions of this group to fail
+    uint64 version = 4;
+
+    // total_weight is the sum of the group members' weights.
+    string total_weight = 5;
+}
+```
+
+Group members are stored in a `groupMemberTable` using the `GroupMember` type:
+
+```proto
+message GroupMember {
+
+    // group_id is the unique ID of the group.
+    uint64 group_id = 1;
+
+    // member is the member data.
+    Member member = 2;
+}
+
+// Member represents a group member with an account address,
+// non-zero weight and metadata.
+message Member {
+
+    // address is the member's account address.
+    string address = 1;
+    
+    // weight is the member's voting weight that should be greater than 0.
+    string weight = 2;
+    
+    // metadata is any arbitrary metadata to attached to the member.
+    bytes metadata = 3;
+}
+```
+
+### Group Account
+
+A group account is an account associated with a group and a decision policy.
+A group account does have a balance.
+
+Group accounts are abstracted from groups because a single group may have
+multiple decision policies for different types of actions. Managing group
+membership separately from decision policies results in the least overhead
+and keeps membership consistent across different policies. The pattern that
+is recommended is to have a single master group account for a given group,
+and then to create separate group accounts with different decision policies
+and delegate the desired permissions from the master account to
+those "sub-accounts" using the [`x/authz` module](adr-030-authz-module.md).
+
+Group accounts are stored as part of the `groupAccountTable` and modeled by `GroupAccountInfo`.
+
+```proto
+message GroupAccountInfo {
```

Comment:
Why do we have `Info` suffix? Based on the paragraph above, I would call this `PolicyAccount`.

Reply:
It was in order to be consistent with x/ecocredit from regen-ledger which uses this suffix too, but I'd be ok with dropping it.
We use `Group` prefix because it's related to a given group and I believe it feels more natural as part of the `x/group` module.

Reply:
So, how about `PolicyGroup`?

Reply:
But then you lose info about the fact that it's an actual account. I kinda like the current `GroupAccount` naming but don't have a strong preference either. It'd be good to have some thoughts from @aaronc who's the primarily designer of the group module afaik.

Reply:
Then, how about `PolicyAccount`? I don't think we need to put `Group` everywhere.